app/Services/Import/Blocks/ECamp3/ECamp3BlockOverviewParser.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Services\Import\Blocks\ECamp3;
+
+use App\Services\Import\Blocks\BlockListParser;
+use Smalot\PdfParser\Parser;
+use Illuminate\Support\Collection;
+
+class ECamp3BlockOverviewParser implements BlockListParser
+{
+    /**
+     * Parses the given PDF file and extracts block data.
+     *
+     * @param string $filePath
+     * @return Collection
+     */
+    public function parse(string $filePath): Collection
+    {
+        // Load PDF
+        $parser = new Parser();
+        $pdf = $parser->parseFile($filePath);
+
+        // Extract raw text with positions
+        $pages = $pdf->getPages();
+        $text = '';
+
+        foreach ($pages as $page) {
+            $objs = $page->getDataTm();
+            foreach ($objs as $obj) {
+                // Add spaces manually between words
+                $text .= ' ' . $obj[1];
+            }
+        }
+
+        $text = mb_convert_encoding($text, 'UTF-8', 'auto');
+        return $this->extractBlocks($text);
+    }
+
+    /**
+     * Extract block data from the text.
+     *
+     * @param string $text
+     * @return Collection
+     */
+    private function extractBlocks(string $text): Collection
+    {
+        $blocks = collect();
+        $pattern = '/.+?\s+(\d+)\.(\d+)\s+(.+?)\s+[A-Za-z]{2}\s+(\d{1,2})\.(\d{1,2})\.(\d{4})\s+(\d{2}:\d{2})\s*-\s*(\d{2}:\d{2})/u';
```

Comment:
`+?` könnte man einfacher als `*` schreiben oder?

Reply:
"?" ist für lazy matching wichtig. + ist 1-unendlich, * ist 0-unendlich, wenn ich es richtig im Kopf habe. ohne das "?" wird in die erste Gruppe alles bis zum letzten Block gematched.

Reply:
Wenn du beim Datum `\.` durch `(\.|\/)` ersetzt müsste es mit französischen Daten auch klappen. Zwar nicht mit englisch, weil dort Monat und Tag vertauscht sind. Aber auch hier kann man argumentieren: Qualix unterstützt selber kein englisch, darum kann es auch kein englisches eCamp v3 PDF verarbeiten.

Reply:
Baue diesen Teil gerade um. Mir ist noch aufgefallen, dass es einen Unterschied zwischen Deutsch und Deutsch (Pfadi) gibt... bei der ersten Sprache gibt es keinen Punkt nach dem Wochentag-Kürzel, bei der zweiten schon 🙈